backends/smt2/smtio.py

```diff
@@ -232,17 +232,21 @@ def setup(self):
             if self.logic_uf: self.logic += "UF"
             if self.logic_bv: self.logic += "BV"
             if self.logic_dt: self.logic = "ALL"
+            if self.solver == "yices" and self.forall: self.logic = "BV"
 
         self.setup_done = True
 
-        for stmt in self.info_stmts:
-            self.write(stmt)
+        if self.forall and self.solver == "yices":
+            self.write("(set-option :yices-ef-max-iters 1000000000)")
 
         if self.produce_models:
             self.write("(set-option :produce-models true)")
 
         self.write("(set-logic %s)" % self.logic)
 
+        for stmt in self.info_stmts:
+            self.write(stmt)
```

Comment:
Why move the info statements to after `set-logic`?

Reply:
I was having trouble with yices requiring `(set-logic)` before any types were used (e.g. `BitVec` in `(define-sort |demo9_s| () (_ BitVec 17))`) and so tried to put `(set-logic)` as high up as possible.  But you're right that this is unnecessary and undesirable, and I will move it back.